### PR TITLE
Feature/cmr 7487 - checking Content Type->Type->Subtype together

### PR DESCRIPTION
**Pull request by jceaser**
This tick is not ready for open review, still working with Bamboo to get a green build

This change adds a new required ingest rule to check that the Related URL Content Type, Type, and Subtype all match a current value in KMS. Keywords which are valid but out of context will be rejected and cause collections to not load.

### CODE REVIEW

File: ingest-app/src/cmr/ingest/validation/validation.clj [hunk 1]
```diff
@@ -90,6 +90,54 @@
...
+   (v/every
+    [(datacenter-url-validator kms-index)
+     (datacenter-contact-url-validator kms-index :ContactGroups)
+     (datacenter-contact-url-validator kms-index :ContactPersons)])})
+
+(defn manditory-keyword-validations
+  "The list of keywords which are validated if the optional validation is not
+   requested. This function is in contrast to keyword-validations which represent
+   the list of validations to use when reqesting validation. Over time, buisness
+   rules will dictate moving move values from that function to this one as
```
Comment (sxu123) [inline ingest-app/src/cmr/ingest/validation/validation.clj:134]: typos:  "move" to more;  "manditory" to mandatory.  
Should mandatory-keyword-validations be performed in keyword-validations as well?

Comment (jceaser) [reply ingest-app/src/cmr/ingest/validation/validation.clj:134]: I did not want to rename the original function, but I will to make things more clear

File: dev-system/resources/kms_examples/rucontenttype [hunk 2]
```diff
@@ -1,75 +1,99 @@
...
-"VIEW RELATED INFORMATION","PUBLICATIONS","13a4deec-bd22-4864-9804-77fac181f484"
-"VIEW RELATED INFORMATION","READ-ME","aa3cea98-b20a-4de8-8f22-7a8b30784625"
-"VIEW RELATED INFORMATION","REQUIREMENTS AND DESIGN","86b8b121-d710-4c5b-84b0-7b40717f6c76"
-"VIEW RELATED INFORMATION","SCIENCE DATA PRODUCT SOFTWARE DOCUMENTATION","e8e6e972-832f-4501-a721-4108f33332d6"
-"VIEW RELATED INFORMATION","SCIENCE DATA PRODUCT VALIDATION","15b0a4c4-b39d-48f5-92d2-905e45e6dc6a"
-"VIEW RELATED INFORMATION","USER FEEDBACK PAGE","ab2fce71-e5f9-4ba6-bfb1-bc428a8b7dd8"
-"VIEW RELATED INFORMATION","USER'S GUIDE","d1996d91-e824-4b24-b94e-3aae4543b63b"
-"VIEW RELATED INFORMATION","","5ec1bb9d-0efc-4099-9b31-ec791bbd8145"
+"Hits: 98","page_num: 1","page_size: 2000","Keyword Version: draft","Revision: 2021-07-30 10:51:26","Timestamp: 2021-07-30 10:58:11","Terms Of Use: https://cdn.earthdata.nasa.gov/conduit/upload/5182/KeywordsCommunityGuide_Baseline_v1_SIGNED_FINAL.pdf","The most up to date XML representations can be found here: https://gcmd.sit.earthdata.nasa.gov/kms/concepts/concept_scheme/rucontenttype/?format=xml""Case native"
```
Comment (eereiter) [inline dev-system/resources/kms_examples/rucontenttype:1]: Get rid of "Hits: 98","page_num: 1","page_size: 2000",

File: ingest-app/docs/api.md [hunk 3]
```diff
@@ -4,54 +4,60 @@ See the [CMR Data Partner User Guide](https://wiki.earthdata.nasa.gov/display/CM
...
-    * [DELETE - Delete a variable.](#delete-variable)
-  * /providers/\<provider-id>/services/\<native-id>
-    * [PUT - Create or update a service.](#create-update-service)
-    * [DELETE - Delete a service.](#delete-service)
-  * /providers/\<provider-id>/tools/\<native-id>
-    * [PUT - Create or update a tool.](#create-update-tool)
-    * [DELETE - Delete a tool.](#delete-tool)
-  * /providers/\<provider-id>/subscriptions
+* /providers/\<provider-id>/validate/collection/\<native-id>
+	* [POST - Validate collection metadata.](#validate-collection)
```
Comment (eereiter) [inline ingest-app/docs/api.md:20]: The spacing is off on this one.

File: ingest-app/src/cmr/ingest/validation/validation.clj [hunk 4]
```diff
@@ -121,9 +169,11 @@
        (match-kms-keywords-validation
         kms-index :granule-data-format msg/data-format-not-matches-kms-keywords)}
      :RelatedUrls
-      (v/every {:GetData {:Format (match-getdata-format-kms-keywords-validation
-                                   kms-index
-                                   msg/getdata-format-not-matches-kms-keywords)}})}))
+      [(match-related-url-kms-keywords-validations kms-index)
+        (datacenter-url-validators kms-index)
```
Comment (eereiter) [inline ingest-app/src/cmr/ingest/validation/validation.clj:173]: Spacing is off in this function.

**Review by sxu123 — APPROVED** [review]

File: dev-system/resources/kms_examples/rucontenttype (hunk 2 above)
Comment (jceaser) [reply dev-system/resources/kms_examples/rucontenttype:1]: KMS does this now for schemes which have not been "grandfathered", which is all the short lists. I feel more comfortable knowing that CMR can handle (and ignore) this new content. Also, if there are more then 2000 related urls, then KMS will page them, CMR does not support this yet, but it is something we should look to fix.

**Review by eereiter — APPROVED** [review]

**Pull request merged**